src/app/renderer.js

```diff
@@ -61,6 +61,9 @@ function Renderer (editor, executionContext, updateFiles, transactionDebugger) {
       });
     }
 
+    vm.stateManager.revert(function () {
+      vm.stateManager.checkpoint();
```

Comment:
Is revert rolling back *and* removing the checkpoint that's why you need/want to add it again?